datmo/core/util/misc_functions.py

```diff
@@ -85,3 +85,15 @@ def create_unique_hash(base_hash=None, salt=None):
                           datetime.datetime(1970, 1, 1)).total_seconds() * 100000
     sha1.update(salt+str(timestamp_microsec).encode('utf-8'))
     return sha1.hexdigest()
+
+def mutually_exclusive(mutually_exclusive_args, arguments_dictionary, dictionary):
+    mutually_exclusive_arg_count = 0
+    for arg in mutually_exclusive_args:
+        if arg in arguments_dictionary and arguments_dictionary[arg]:
```

Comment:
shouldn't this be `arguments_dictionary[arg] != None`

Reply:
All variables in the snapshot entity has different default values. Hence, let's suppose `commit_id` and `code_id`, would have values such as `arguments_dictionary['code_id'] = 'example_code_id'` and `arguments_dictionary['commit_id'] = ''`.

This would create a random `dictionary` based what key comes at the last, such as `dictionary['commit_id']=''`